engine/crates/sdl-parser/src/lib.rs

```diff
@@ -144,8 +165,79 @@ pub async fn parse<'a>(
         directives.to_definition(),
     );
 
-    let schema = parse_schema(schema)?;
+    grafbase_engine::parser::parse_schema(schema)
+}
 
+pub fn parse_used_directives(schema: &str) -> grafbase_engine::parser::Result<UsedDirectives> {
+    let doc = parse_schema(schema)?;
+    let mut used = UsedDirectives::default();
+    fn directive_name(directive: Positioned<ConstDirective>) -> String {
+        directive.node.name.node.to_string()
```

Comment:
Up to you, but what do you think about putting this stuff in it's own module & just `pub use`ing it from the root?

Reply:
fair point

Reply:
done